### ISSUE TITLE
Authentication for Google Docker registry

### ISSUE THREAD

**Issue by nhammond** [StanfordBioinformatics/loom#265]
To access a registry using the service account *.json key, we need to first execute "docker login" and provide the service account email address and key.

[https://cloud.google.com/container-registry/docs/advanced-authentication](https://cloud.google.com/container-registry/docs/advanced-authentication)

```
docker login -e 1234@5678.com -u _json_key -p "$(cat keyfile.json)" https://gcr.io
```

This raises a deprecation warning: "Flag --email has been deprecated, will be removed in 1.13."

**Comment by nhammond** [comment]
This is an alternative method:

docker login -u oauth2accesstoken -p "$(gcloud auth print-access-token)" https://gcr.io

**Comment by nhammond** [comment]
This may be useful:
https://github.com/GoogleCloudPlatform/docker-credential-gcr

**Comment by isaacliao** [comment]
Implemented in 7a4c9dcb8fce90c0bcd6a44dd89f6071d14901bf
